lib/component/catalog_table.es6.js

```diff
@@ -87,29 +105,27 @@ function catalogTableController($scope, api) {
   // is filtered by release options and search keyword (if exists).
   $scope.downloadCSV = function() {
     let filename = 'result.csv';
-    let releaseKeys = getReleaseKeys(ctrl.releases);
-    let csv = apiMatrix.matrixToCSV(releaseKeys,
-      $scope.apiCatalogMatrix);
-    if (csv === null) return;
-    if (!csv.match(/^data:text\/csv/i)) {
-      csv = 'data:text/csv;charset=utf-8,' + csv;
-    }
-    let data = encodeURI(csv);
-    let link = document.createElement('a');
-    link.setAttribute('href', data);
-    link.setAttribute('download', filename);
-    link.click();
+    let releaseKeys = getReleaseKeys(ctlr.releases);
+    matrixCtlr.getMatrixCSV(releaseKeys).then(csv => {
+      if (csv === null) return;
+      if (!csv.match(/^data:text\/csv/i)) {
+        csv = 'data:text/csv;charset=utf-8,' + csv;
```

Comment:
Are there usages of getMatrixCSV that don't want this prefix?

Reply:
Not at the moment, but there may be in the future. We may expose the CSV data over a REST API.